soh/src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -1330,7 +1343,17 @@ void EnOssan_GiveItemWithFanfare(GlobalContext* globalCtx, EnOssan* this) {
     Player* player = GET_PLAYER(globalCtx);
 
     osSyncPrintf("\n" VT_FGCOL(YELLOW) "初めて手にいれた！！" VT_RST "\n\n");
-    func_8002F434(&this->actor, globalCtx, this->shelfSlots[this->cursorIndex]->getItemId, 120.0f, 120.0f);
+    if (!gSaveContext.n64ddFlag) {
+        func_8002F434(&this->actor, globalCtx, this->shelfSlots[this->cursorIndex]->getItemId, 120.0f, 120.0f);
+    } else {
+        ShopItemIdentity shopItemIdentity = Randomizer_IdentifyShopItem(globalCtx->sceneNum, this->cursorIndex);
+        if (shopItemIdentity.randomizerCheck != RC_UNKNOWN_CHECK) {
```

Comment:
this logic isn't quite clear, a comment explaining why we might end up with an unknown check here and why the correct response to that is doing what we do in vanilla could be nice